src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -2872,14 +2872,22 @@ trait Types
 
     override def paramTypes = mapList(params)(symTpe) // OPT use mapList rather than .map
 
+    final def resultTypeOwnParamTypes: Type = {
+      if (isTrivial || phase.erasedTypes) resultType
+      else resultType0(paramTypes)
+    }
     override def resultType(actuals: List[Type]) =
       if (isTrivial || phase.erasedTypes) resultType
-      else if (/*isDependentMethodType &&*/ sameLength(actuals, params)) {
+      else resultType0(actuals)
+
+    private def resultType0(actuals: List[Type]): Type = {
+      if (/*isDependentMethodType &&*/ sameLength(actuals, params)) {
         val idm = new InstantiateDependentMap(params, actuals)
         val res = idm(resultType).deconst
         existentialAbstraction(idm.existentialsNeeded, res)
       }
       else existentialAbstraction(params, resultType)
+    }
```

Comment:
I wonder if there is a rule for which methods get braces. `resultTypeOwnParamTypesBraces` would obvi get braces.